### ISSUE TITLE
Can you please clarify the logic behind "one-hot encoding the labels to use MSE"

### ISSUE THREAD

**Issue by worldpeaceaspirer** [tensorflow/tensorflow#49465]
## URL(s) with the issue:

https://www.tensorflow.org/guide/keras/train_and_evaluate/#custom_losses

Thank you for the very detailed explanation about Train and Evaluate but I have a query. Raised a [Stack Overflow question](https://stackoverflow.com/questions/67638773/can-you-please-clarify-the-logic-behind-one-hot-encoding-the-labels-to-use-mse) as well.

## Description of issue (what needs changing):
Nothing needs a change but I want to understand the statement,

> We need to one-hot encode the labels to use MSE

and want to understand exactly, how **`One-Hot-Encoding`** and **`MSE`** work together (because **`One-Hot-Encoding`** just creates columns with 0's and a 1), in the code mentioned below in the [Documentation of Keras Train and Evaluate ](https://www.tensorflow.org/guide/keras/train_and_evaluate/#custom_losses)

```python
def custom_mean_squared_error(y_true, y_pred):
    return tf.math.reduce_mean(tf.square(y_true - y_pred))


model = get_uncompiled_model()
model.compile(optimizer=keras.optimizers.Adam(), loss=custom_mean_squared_error)

# We need to one-hot encode the labels to use MSE
y_train_one_hot = tf.one_hot(y_train, depth=10)
model.fit(x_train, y_train_one_hot, batch_size=64, epochs=1)
```

**Comment by tilakrayal** [comment]
@worldpeaceaspirer ,

This is not a TensorFlow bug or feature request, it is better asked on [StackOverflow](https://stackoverflow.com/questions/tagged/tensorflow). There is a larger community that reads questions there. Thanks!

Please refer to this links for more information about one-hot encoding.[link1](https://towardsdatascience.com/building-a-one-hot-encoding-layer-with-tensorflow-f907d686bf39),[link2](https://www.tensorflow.org/tutorials/text/word_embeddings#one-hot_encodings)

Thanks!

**Comment by Saduf2019** [comment]
@worldpeaceaspirer 
The Adam Optimizer can, in theory,minimize MSE loss for a one-hot-encoded vector,there is no mathematical barrier,but that does not mean it is the best choice in this case.
The link demonstrates custom loss functions and not teach you ML.
you do not need to apply OHE if your application is classification you can use sparse categorical entropy loss on labelencoded data.

As mentioned above, please create a stackoverflow issue for any further queries, and move this to closed status.
